apps/scraper-runner/src/main.ts

```diff
@@ -18,8 +20,27 @@ function getTopPercentage<T>(arr: T[], percent = 5): T[] {
   return arr.slice(0, Math.floor((arr.length * percent) / 100));
 }
 
+function createFilename(impacted: string, recommendation: string): string {
+  return `${impacted}-${recommendation}.json`.replace('/', '_');
+}
+
+function createImpactList(healthRemedies: HealthRemedies[]): ImpactFileList {
+  return healthRemedies.map((healthRemedy) => {
+    return {
+      impacted: healthRemedy.impacted,
+      recommendations: healthRemedy.recommendations.map((recommendation) => {
+        return {
+          fileName: createFilename(healthRemedy.impacted, recommendation),
+          recommendation,
+        };
+      }),
+    };
```

Comment:
Why not have the HealthRemedies type include 
`impacted: string,
recommendations: []RecommendationInfo`
where RecommendationInfo is of type
`filename: string,
recommendation: string`

When the HealthRemedy object is first created, its okay to only populate the impacted and recommendationInfo.recommendation field

Then instead of doing these mappings here, you can just loop through your healthremedies list and for every remedy, assign the recommendationInfo.filename field

Reply:
Oh ok, that makes so much more sense.

I made all the changes for using RecommendationInfo